R/utilities-simulation.R

```diff
@@ -244,17 +236,94 @@ createSimulationBatch <- function(simulation, parametersOrPaths = NULL, molecule
+#' @param simulationRunOptions Optional instance of a \code{SimulationRunOptions} used during the simulation run.
+#' @param silentMode If \code{TRUE}, no warnings are displayed if a simulation fails.
+#' Default is \code{FALSE}.
+#'
+#' @return Nested list of \code{SimulationResults} objects. The first level of the list are the IDs of the simulations of SimulationBatches, containing a list of \code{SimulationResults} for each set of parameter/initial values. If a simulation with a parameter/initial values set fails, the result for this run is \code{NULL}
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' sim <- loadSimulation("simple", loadFromCache = TRUE)
+#' parameters <- c("Organism|Liver|Volume", "R1|k1")
+#' molecules <- "Organism|Liver|A"
+#' # Create two simulation batches. In this case, they are based on the same simulation
+#' # and vary the same paramter and initial values.
+#' simulationBatch1 <- createSimulationBatch(simulation = sim,
+#' parametersOrPaths = parameters,
+#' moleculesOrPaths = molecules)
+#' simulationBatch2 <- createSimulationBatch(simulation = sim,
+#' parametersOrPaths = parameters,
+#' moleculesOrPaths = molecules)
+#' #Ids of run values
+#' ids <- c()
+#' ids[[1]] <- simulationBatch1$addRunValues(parameterValues = c(1, 2), initialValues = 1)
+#' ids[[2]] <- simulationBatch1$addRunValues(parameterValues = c(1.6, 2.4), initialValues = 3)
+#' ids[[3]] <- simulationBatch2$addRunValues(parameterValues = c(4, 2), initialValues = 4)
+#' ids[[4]] <- simulationBatch2$addRunValues(parameterValues = c(2.6, 4.4), initialValues = 5)
+#' res <- runSimulationBatches(simulationBatches = list(simulationBatch1, simulationBatch2))
+#' }
+runSimulationBatches <- function(simulationBatches, simulationRunOptions = NULL, silentMode = FALSE) {
+  validateIsOfType(simulationBatches, "SimulationBatch")
+  simulationRunner <- getNetTask("ConcurrentSimulationRunner")
+  if (!is.null(simulationRunOptions)) {
+    validateIsOfType(simulationRunOptions, SimulationRunOptions)
+    rClr::clrSet(simulationRunner, "SimulationRunOptions", simulationRunOptions$ref)
+  }
+
+  simulationBatches <- c(simulationBatches)
+  # Result Id <-> simulation batch pointer id map to get the correct simulation for the results.
+  # Using the Id of the pointer instead of the Id of the simulation as multiple
+  # SimulationBatches can be created with the same simulation
+  # Each SimulationBatchRunValues has its own id, which will be the id of the result
+  resultsIdSimulationIdMap <- list()
+  # Map of simulations ids to simulations objects
+  simulationIdSimulationMap <- vector("list", length(simulationBatches))
+  # Iterate through all simulation batches
+  for (simBatchIdx in seq_along(simulationBatches)) {
+    simBatch <- simulationBatches[[simBatchIdx]]
+    # Put the simulation into sim pointer id <-> sim map
+    simId <- deparse(rClr::clrGetExtPtr(simBatch$ref))

```

Comment:
WHAT IS THIS? LEt's please have a batch return an id or sthg.

Reply:
:D Yes then we need to add an ID field to `OSPSuite.R.Services.SettingsForConcurrentRunSimulationBatch` in core.